src/App.tsx

```diff
@@ -1,24 +1,195 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
+
 import { UserWarning } from './UserWarning';
+import { Header } from './components/Header/Header';
+import { TodoList } from './components/TodoList/TodoList';
+import { Footer } from './components/Footer/Footer';
+import { Notification } from './components/Notification/Notification';
+
+import { Todo } from './types/Todo';
+
+import {
+  getTodos, addTodo, onDelete, onUpdate,
+} from './api/todos';
 
-const USER_ID = 0;
+const USER_ID = 6476;
+
+export enum SortType {
+  ALL = 'All',
+  ACTIVE = 'Active',
+  COMPLETED = 'Completed',
+}
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterBy, setFilterBy] = useState('All');
+  const [isError, setIsError] = useState(false);
+  const [inputDisable, setInputDisable] = useState(false);
+  const [title, setTitle] = useState('');
+
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const visibleTodos = useMemo(() => {
+    return (todos.filter((todo) => {
+      switch (filterBy) {
+        case SortType.ALL:
+          return todos;
+
+        case SortType.ACTIVE:
+          return !todo.completed;
+
+        case SortType.COMPLETED:
+          return todo.completed;
+
+        default:
+          return [];
+      }
+    })
+    );
+  }, [filterBy, todos]);
+
+  const noCompleteTodos = todos.some((todo) => todo.completed);
+  const isMustBeCompleted = todos.filter((todo) => !todo.completed).length;
+
+  const loadTodosData = async () => {
+    try {
+      const todosFromServer = await getTodos(USER_ID);
+
+      setInputDisable(false);
+      setIsError(false);
+      setTempTodo(null);
+
+      setTodos(todosFromServer);
+    } catch (error) {
+      setIsError(true);
+      setErrorMessage("Can't load data...");
+    }
+  };
+
+  useEffect(() => {
+    loadTodosData();
+  }, []);
+
+  const createNewTodo = (query: string) => {
+    const newId = Math.max(...todos.map((todo) => todo.id + 1));
+
+    const newTodo = {
+      id: newId,
+      userId: USER_ID,
+      title: query,
+      completed: false,
+    };
+
+    return newTodo;
+  };
+
+  const onCompletedChange = (todo: Todo): Todo | void => {
+    onUpdate(todo.id, {
+      id: todo.id,
+      userId: todo.userId,
+      title: todo.title,
+      completed: !todo.completed,
+    });
+
+    setTempTodo(null);
+    loadTodosData();
+  };
+
+  const dataTodo = {
+    userId: USER_ID,
+    title,
+    completed: false,
+  };
+
+  const sendTodo = async () => {
+    setTempTodo({ id: 0, ...dataTodo });
+
+    try {
+      if (title) {
+        await addTodo(createNewTodo(title));
+      } else {
+        setErrorMessage('....');
```

Comment:
why `....`? Set `''` empty string instead
Also, instead of creating `errorMessage` and `isError` states you can keep only `errorMessage` and check it to know if there is any error (empty string is a falsy value)